api/src/ginger_calls.rs

```diff
@@ -1,85 +1,117 @@
+    tweedle::{
+        dum::{
+            Projective as Projective,
+            Affine as Affine
+        },
+        dee::Affine as PCAffine
     },
-}, FromBytes, ToBytes, BigInteger768, ProjectiveCurve, AffineCurve, ToConstraintField, UniformRand, ToBits};
+}, FromBytes, FromBytesChecked, validity::SemanticallyValid,
+   ToBytes, BigInteger256, ProjectiveCurve, AffineCurve, ToConstraintField, UniformRand, ToBits};
 use primitives::{crh::{
-    poseidon::{
-        MNT4PoseidonHash,
-        batched_crh::MNT4BatchPoseidonHash as BatchFieldHash,
-    },
+    poseidon::parameters::tweedle::{TweedleFrPoseidonHash as PoseidonHash, TweedleFrBatchPoseidonHash as BatchFieldHash},
     FieldBasedHash,
     bowe_hopwood::{
         BoweHopwoodPedersenCRH, BoweHopwoodPedersenParameters
     },
 }, merkle_tree::field_based_mht::{
-    smt::{BigMerkleTree, LazyBigMerkleTree, Coord, OperationLeaf},
+    // smt::{BigMerkleTree, LazyBigMerkleTree, Coord, OperationLeaf},
     optimized::FieldBasedOptimizedMHT,
-    poseidon::{MNT4753_PHANTOM_MERKLE_ROOT as PHANTOM_MERKLE_ROOT, MNT4753_MHT_POSEIDON_PARAMETERS as MHT_PARAMETERS},
+    parameters::tweedle_fr::TWEEDLE_MHT_POSEIDON_PARAMETERS as MHT_PARAMETERS,
+    // parameters::mnt4753::{MNT4753_PHANTOM_MERKLE_ROOT as PHANTOM_MERKLE_ROOT, MNT4753_MHT_POSEIDON_PARAMETERS as MHT_PARAMETERS},
     FieldBasedMerkleTree, FieldBasedMerkleTreePrecomputedEmptyConstants,
     FieldBasedMerkleTreeParameters, BatchFieldBasedMerkleTreeParameters,
     FieldBasedMerkleTreePath, FieldBasedBinaryMHTPath,
 }, signature::{
     FieldBasedSignatureScheme, schnorr::field_based_schnorr::{
-        FieldBasedSchnorrSignatureScheme, FieldBasedSchnorrSignature
+        FieldBasedSchnorrSignatureScheme, FieldBasedSchnorrSignature,
+        FieldBasedSchnorrPk,
     },
-}, vrf::{FieldBasedVrf, ecvrf::*}, ActionLeaf};
-use proof_systems::groth16::{
-    Proof, create_random_proof,
-    prepare_verifying_key, verify_proof,
-};
+}, vrf::{FieldBasedVrf, ecvrf::*}/*, ActionLeaf*/};
+
+use marlin::*;
+use blake2::Blake2s;
+use poly_commit::ipa_pc::InnerProductArgPC;
+// use proof_systems::groth16::{
+//     create_random_proof,
+//     prepare_verifying_key, verify_proof,
+// };
 use demo_circuit::{
     constants::{
         VRFParams, VRFWindow,
     },
     naive_threshold_sig::*
 };
 use rand::{
-    SeedableRng, rngs::OsRng
+    SeedableRng, rngs::OsRng, thread_rng
 };
 use rand_xorshift::XorShiftRng;
 
 use std::{
-    fs::File, io::Result as IoResult, path::Path
+    fs::File, io::Result as IoResult/*, path::Path*/
 };
 use lazy_static::*;
 
 pub type FieldElement = Fr;
 
-pub const FIELD_SIZE: usize = 96; //Field size in bytes
-pub const SCALAR_FIELD_SIZE: usize = FIELD_SIZE;// 96
+// #[derive(Clone)]
+// struct MarlinNoLCNoZk;
+//
+// impl MarlinConfig for MarlinNoLCNoZk {
+//     const LC_OPT: bool = false;
+//     const ZK: bool = false;
+// }
+
+type IPAPC = InnerProductArgPC<PCAffine, Blake2s>;
```

Comment:
For types and constants would be nice to do something similar to https://github.com/HorizenOfficial/zendoo-mc-cryptolib/blob/as/ginger_algebra_ocl/src/type_mapping.rs

Reply:
Done